modules/admin-ui/src/main/java/org/opencastproject/adminui/endpoint/AbstractEventEndpoint.java

```diff
@@ -2582,6 +2583,19 @@ public Response getEvents(@QueryParam("id") String id, @QueryParam("commentReaso
             return Response.status(SC_BAD_REQUEST).build();
         }
       }
+      if (EventListQuery.FILTER_IS_PUBLISHED_NAME.equals(name)) {
+        switch (IsPublished.valueOf(filters.get(name))) {
+          case YES:
+            query.withIsPublished(true);
+            break;
+          case NO:
+            query.withIsPublished(false);
+            break;
+          default:
+            logger.info("Unknown is published status {}", filters.get(name));
+            return Response.status(SC_BAD_REQUEST).build();
+        }
+      }
```

Comment:
Logically what you've done here makes sense, but Java throws this if you pass something *other* than `true`/`false`:
```
java.lang.IllegalArgumentException: No enum constant org.opencastproject.index.service.resources.list.provider.EventsListProvider.IsPublished.
FALSE
        at java.lang.Enum.valueOf(Enum.java:273) ~[?:?]
        at org.opencastproject.index.service.resources.list.provider.EventsListProvider$IsPublished.valueOf(EventsListProvider.java:68) ~[?:?]

```

So let's check if the value of the filter is valid, throw a 400 if it's not, and then handle the case where `isPublished` is unspecified.

```suggestion
      if (EventListQuery.FILTER_IS_PUBLISHED_NAME.equals(name)) {
        if (filters.containsKey(name)) {
          switch (IsPublished.valueOf(filters.get(name))) {
            case YES:
              query.withIsPublished(true);
              break;
            case NO:
              query.withIsPublished(false);
              break;
            default:
              break;
          }
        } else {
          logger.info("Query for invalid published status: {}", filters.get(name));
          return Response.status(SC_BAD_REQUEST).build();
        }
      }
```

And yes, I'm morbidly aware that just a few lines above this we do *exactly the same wrong thing*

Reply:
I have now applied your suggested changes and thanks, I actually did not knew that this could happen :) 
Also sorry for taking so long to reply.